src/aro/Parser.zig

```diff
@@ -260,6 +262,26 @@ fn validateExtendedIdentifier(p: *Parser) !bool {
         if (!warned) {
             warned = try checkIdentifierCodepointWarnings(p.comp, codepoint, loc);
         }
+
+        // Check NFC normalization.
+        if (!normalized) continue;
+        if (codepoint >= 0x10000 and codepoint <= 0x10FFFF) continue;
```

Comment:
Is this right? I think I made the same mistake - if you adapted this from the example in the unicode docs, it's written for Java which is UTF-16:

```
    int ch = source.codepointAt(i);
        if (Character.isSupplementaryCodePoint(ch)) ++i;
   ...
```

So it's not skipping over supplementary codepoints; it's just skipping the low surrogate of the UTF-16 encoding.

Reply:
Oh right, thanks!